common/scala/src/main/scala/org/apache/openwhisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -132,6 +132,10 @@ class KafkaConsumerConnector(
         } else {
           throw e
         }
+      case e: WakeupException =>
+        logging.info(this, s"WakeupException happened when do commit action for topic ${topic}")
+        close()
```

Comment:
It seems this just leave the consumer closed and there is no logic to recreate a consumer.
Since it propagates the exception to `MessageFeed`, `MessageFeed` will keep failing.

I think we need to 1. catch the exception here and recreate the consumer or 2. throw the exception, catch it in `MessageFeed`, and recreate the consumer.
The same approach should be taken here as well: https://github.com/apache/incubator-openwhisk/pull/4459/files#diff-8bf7d87d98238d39ba7f36d2c52def6fR202

I am inclined to handle this exception in `KafkaConsumerConnector` layer because the same approach is already taken here: https://github.com/apache/incubator-openwhisk/pull/4459/files#diff-8bf7d87d98238d39ba7f36d2c52def6fR115

So in case of error, the consumer will be recreated without notifying `MessageFeed`.

Reply:
In normal situation, when one consumer is closed, we need to create a new one.

But here, the user has a clear purpose: `close the consumer gracefully using consumer.wakeup` and have no purpose to create a new consumer again.

Reply:
`consumer.wakeup` is only used to prevent infinite hangs: https://github.com/apache/incubator-openwhisk/pull/4459/files#diff-8bf7d87d98238d39ba7f36d2c52def6fR73
Even there is no public method for `wakeup()` in `KafkaConsumerConnector`.

Is there any part that a user explicitly close this consumer using `wakeup()`?

Reply:
Oh, in current github master branch, has no any part that user explicitly close this consumer using wakeup().

Maybe one day in future, for some special scene, may expose a public method `wakup()` in KafkaConsumerConnector.

I added this patch,  just for  `future expansion`.

Reply:
Then how about keeping the current behavior(recreating the consumer) and update it accordingly with the enhancement in the future?

Reply:
Good idea!